src/main/java/org/jenkinsci/test/acceptance/junit/WithPlugins.java

```diff
@@ -45,15 +45,29 @@
  *
  * The latter example declares that running the test with older version is pointless, typically because of missing feature.
  *
- * The annotation guarantees that the plugin is installed in required or later version.
+ * In normal mode the annotation guarantees that the plugin is installed in required or later version.
  * If the plugin is already installed but not in correct version then
  * the environment variable NEVER_REPLACE_EXISTING_PLUGINS is evaluated:
  * <ul>
  *     <li>if the environment variable is set then the test will be skipped.</li>
  *     <li>if the environment variable is undefined then the installed version
  *     of the plugin is overwritten with the latest version of the plugin. If
  *     required version is not available in update center, the test will fail.</li>
-*  </ul>
+ *  </ul>
+ *
+ * There is also a pre configured plugins mode, running in this mode means that the ATH is using a war file that (somehow)
```

Comment:
Once we kill `NEVER_REPLACE_EXISTING_PLUGINS` this will be the "normal" mode

Reply:
If we make this the normal mode it will break all existing ATH runs out there but the essentials ones and maybe some of those using `PLUGINS_DIR`, better to make this opt-in for the moment, IMO

Reply:
Sure, what I meant is I do not like distinguishing the *modes*. It is just an option that will cause test to skip/fail under certain situation. We may think of at as modes of execution but then there is not much to say about the normal mode. I do not feel strongly here.